homeassistant/loader.py

```diff
@@ -1132,43 +1130,39 @@ def bind_hass(func: _CallableT) -> _CallableT:
 
 async def _async_component_dependencies(
     hass: HomeAssistant,
-    start_domain: str,
     integration: Integration,
-    loaded: set[str],
-    loading: set[str],
 ) -> set[str]:
-    """Recursive function to get component dependencies.
+    """Async function to get component dependencies."""
+    loading = set()
+    loaded = set()
 
-    Async friendly.
-    """
-    domain = integration.domain
-    loading.add(domain)
+    async def helper(integration: Integration) -> None:
+        """Recursive async helper to get component dependencies."""
```

Comment:
`helper` is not a good name.

Maybe call it `component_dependencies_impl` since this is the actual implementation and `_async_component_dependencies` is just a wrapper.